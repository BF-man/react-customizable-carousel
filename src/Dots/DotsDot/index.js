import React, { PureComponent } from 'react'
import FaCircle from 'react-icons/lib/fa/circle'
import FaCircleO from 'react-icons/lib/fa/circle-o'

export class DotsDot extends PureComponent {
  handleClick = () => {
    const { onClick, index } = this.props
    onClick(index)
  }
  get cssClass () {
    const { className } = this.props
    return `rcc-Dot ${className || ''}`
  }
  render () {
    const { current, component } = this.props
    const CustomContent = component
    return (
      <span className={this.cssClass} onClick={this.handleClick}>
        {component ? <CustomContent current={current} /> : <DefaultContent current={current} />}
      </span>
    )
  }
}

const DefaultContent = ({ current }) => current ? <FaCircle /> : <FaCircleO />
