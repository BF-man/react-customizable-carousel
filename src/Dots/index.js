import React, { PureComponent } from 'react'
import { DotsDot } from './DotsDot'

export class Dots extends PureComponent {
  get cssClass () {
    const { className } = this.props
    return `rcc-Dots ${className || ''}`
  }
  render () {
    const { onDotClick, currentIndex, itemsCount, dotComponent, dotWrapperClassName } = this.props
    const indexes = [...Array(itemsCount).keys()]
    return (
      <div className={this.cssClass} >
        {indexes.map(index => (
          <DotsDot
            key={index}
            index={index}
            component={dotComponent}
            onClick={onDotClick}
            current={currentIndex === index}
            className={dotWrapperClassName}
          />
        ))}
      </div>
    )
  }
}
