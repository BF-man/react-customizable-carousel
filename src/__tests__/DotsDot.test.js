import React from 'react'
import renderer from 'react-test-renderer'
import { DotsDot } from '../Dots/DotsDot'

test('DotsDot renders default component', () => {
  const component = renderer.create(
    <DotsDot />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('DotsDot renders custom component', () => {
  const component = renderer.create(
    <DotsDot component={() => <div>Custom component</div>} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('DotsDot adds customClassName ', () => {
  const component = renderer.create(
    <DotsDot className='customClassName' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
