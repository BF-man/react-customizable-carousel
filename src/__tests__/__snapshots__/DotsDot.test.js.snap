// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotsDot adds customClassName  1`] = `
<span
  className="rcc-Dot customClassName"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m20.1 7.9q-3.3 0-6.1 1.6t-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1-1.6-6.1-4.5-4.4-6.1-1.6z m17.2 12.1q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
      />
    </g>
  </svg>
</span>
`;

exports[`DotsDot renders custom component 1`] = `
<span
  className="rcc-Dot "
  onClick={[Function]}
>
  <div>
    Custom component
  </div>
</span>
`;

exports[`DotsDot renders default component 1`] = `
<span
  className="rcc-Dot "
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m20.1 7.9q-3.3 0-6.1 1.6t-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1-1.6-6.1-4.5-4.4-6.1-1.6z m17.2 12.1q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
      />
    </g>
  </svg>
</span>
`;
